crates/test-collector/src/lib.rs

```diff
@@ -556,16 +557,31 @@ pub fn collect_tests(
         })
         .collect_vec()
         .into_iter()
-        .map(|(test_name, config)| TestCase {
-            name: test_name,
-            available_gas: config.available_gas,
-            ignored: config.ignored,
-            expected_result: config.expected_result,
-            fork_config: config.fork_config,
-            fuzzer_config: config.fuzzer_config,
+        .map(|(test_name, config)| {
+            if config.available_gas.is_some() {
+                gas_error = Some(Err(
+                    anyhow!
+                        ("{} - Gas calculation used for available_gas is incorrect, because it doesn't include the cost of contract execution.",
```

Comment:
This is a bad error message, we should just state that it is not supported without bothering the user with unnecessary details

Reply:
Agreed